core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/configuration/script/ScriptConfiguration.java

```diff
@@ -260,18 +337,40 @@ public boolean exists(String scriptName) {
         return scripts;
     }
 
+    public List<Script> getAllDeleted() {
+        List<Script> scripts = new ArrayList<>();
+        CachedRowSet crsScript = getMetadataRepository().executeQuery(
+                FETCH_ALL_DELETED_QUERY,
+                "reader");
+
+        try {
+            while (crsScript.next()) {
+                String scriptId = crsScript.getString("SCRIPT_ID");
+                List<ScriptVersion> scriptVersions = ScriptVersionConfiguration.getInstance().getDeletedByScriptId(scriptId);
+                for (ScriptVersion scriptVersion : scriptVersions) {
+                    getDeleted(new ScriptKey(scriptId, scriptVersion.getNumber())).ifPresent(scripts::add);
+                }
+            }
+
+        } catch (SQLException e) {
+            StringWriter stackTrace = new StringWriter();
+            e.printStackTrace(new PrintWriter(stackTrace));
+
+            log.info(String.format("exception=%s", e));
+            log.debug(String.format("exception.stacktrace=%s", stackTrace));
+        }
+        return scripts;
+    }
+
     @Override
     public void delete(ScriptKey scriptKey) {
-        log.trace(MessageFormat.format("Deleting script {0}", scriptKey.toString()));
```

Comment:
get deletion timestamp here and pass it to ScriptVersionConfiguration#delete and ScriptConfiguation#getDeleteStatement